src/components/FetchAPI.js

```diff
@@ -0,0 +1,63 @@
+import React, { Component } from 'react';
+
+export default class DataFetchingComponent extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      loading: true,
+      error: null,
+      data: [],
+    };
+  }
+
+  componentDidMount() {
+    fetch('https://api.api-ninjas.com/v1/quotes?category=dating', {
+      method: 'GET',
+      headers: {
+        'X-Api-Key': '9m7PseExHusn5wNJIgKbQg==yaobg8xRyBMMWoYM',
+      },
+      contentType: 'application/json',
+    })
+      .then((res) => res.json())
+      .then((data) => {
+        this.setState({
+          loading: false,
+          data,
+        });
+      })
+      .catch((error) => {
+        this.setState({
+          loading: false,
+          error,
+        });
+      });
+  }
```

Comment:
- [ ] Excellent work on successfully implementing the API integration milestone. Your code effectively retrieves data from the API and appears to be functioning smoothly. However, I'd like to point out that your project currently doesn't align with all of the specified requirements.

   - In the project guidelines, there's a specific requirement concerning the utilization of the `useEffect` hook. As I examined your code, I noticed that this hook isn't being utilized. To ensure your project meets all the outlined criteria, I kindly recommend incorporating the `useEffect` hook appropriately.

   - This adjustment will not only bring your project in line with the expectations but also contribute to its overall structure and adherence to best practices. See below for more details on project requirements:
<img width="684" alt="Screenshot 2023-08-11 at 18 08 09" src="https://github.com/MohamedRoshdy2021/Math-Magicians/assets/107192076/0b923825-7e7c-4e0d-9367-b2c952706d57">


Happy coding! 😊🚀🍀